crates/hyperswitch_domain_models/src/api.rs

```diff
@@ -59,20 +59,26 @@ pub struct PaymentLinkStatusData {
 }
 
 #[derive(Debug, Eq, PartialEq)]
-pub enum GenericLinks {
+pub struct GenericLinks {
+    pub allowed_domains: Option<HashSet<String>>,
```

Comment:
this shouldn't be optional since hashset is a required field in business profile